### PR TITLE
Tweaked Fortify Parser To Handle Missing Code Snippet For Finding

### PR DESCRIPTION
**Pull request by ibcoleman**
Fixed issue with Fortify parser where the import will fail if there is no Primary.Snippet value for a given finding. Added issue id and exception handling during parsing so we know where the import failed. In some cases (ok, in my case) <Primary> stanza doesn't have a <Snippet>.

### CODE REVIEW

File: dojo/tools/fortify/parser.py [hunk 1]
```diff
@@ -62,26 +62,36 @@ def __init__(self, filename, test):
...
+                    source = {
+                        "FileName": issue.find("Source").find("FileName").text,
+                        "FilePath": issue.find("Source").find("FilePath").text,
+                        "LineStart": issue.find("Source").find("LineStart").text,
+                        "Snippet": issue.find("Source").find("Snippet").text}
+                    details["Source"] = source
+
+                issue_map.update({issue.attrib['iid']: details})
+        except AttributeError:
+            print("XML Parsing blew up on issue number:", issue_id)
```
Comment (valentijnscholten) [inline dojo/tools/fortify/parser.py:93]: can you use logger.warn or similar?

**Review by valentijnscholten — CHANGES_REQUESTED** [review]

File: dojo/tools/fortify/parser.py (hunk 1 above)
Comment (ibcoleman) [reply dojo/tools/fortify/parser.py]: I think flake-8-your-pr just answered that question:

```
dojo/tools/fortify/parser.py#L13
expected 2 blank lines, found 1 (E302)
```

Comment (valentijnscholten) [reply dojo/tools/fortify/parser.py]: thanks. final request to make the message a bit more professional...

File: dojo/tools/fortify/parser.py [hunk 2]
```diff
@@ -62,26 +65,36 @@ def __init__(self, filename, test):
...
+                    source = {
+                        "FileName": issue.find("Source").find("FileName").text,
+                        "FilePath": issue.find("Source").find("FilePath").text,
+                        "LineStart": issue.find("Source").find("LineStart").text,
+                        "Snippet": issue.find("Source").find("Snippet").text}
+                    details["Source"] = source
+
+                issue_map.update({issue.attrib['iid']: details})
+        except AttributeError:
+            logger.warning("XML Parsing blew up on issue number:", issue_id)
```
Comment (madchap) [inline dojo/tools/fortify/parser.py:96]: Would this line be missing a `%s` and `%` as the placeholder for `issue_id`?

**Comment by ibcoleman** [comment]
Sadly this was a time boxed effort on my part and (as you can see) Python
is not my primary language. Good luck!

Best,

Ian

**Review by valentijnscholten — DISMISSED** [review]

**Comment by valentijnscholten** [comment]
Spent a minute to fix it, so it's good now.

**Comment by madchap** [comment]
> Spent a minute to fix it, so it's good now.

Spent the other minute ;-)

**Review by madchap — APPROVED** [review]

**Comment by valentijnscholten** [comment]
@madchap read this https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html

**Comment by madchap** [comment]
Whatever :-)

**Review by valentijnscholten — APPROVED** [review]
amazing, tears in my eyes.

**Pull request merged**